File: cslacs10/CSLA.Test/Children.vb
<Serializable()> _
Public Class Children
  Inherits BusinessCollectionBase

  Default Public ReadOnly Property Item(ByVal index As Integer) As Child
    Get
      Return CType(list(index), Child)
    End Get
  End Property

  Public Sub Add(ByVal Data As String)
    list.Add(Child.NewChild(Data))
  End Sub

  Public Sub Remove(ByVal child As Child)
    list.Remove(child)
  End Sub

  Friend Shared Function NewChildren() As Children
    Return New Children()
  End Function

  Friend Shared Function GetChildren(ByVal dr As IDataReader) As Children
    ' TODO: load child data
  End Function

  Friend Sub Update(ByVal tr As IDbTransaction)
    Dim child As child

    For Each child In list
      child.Update(tr)
    Next
  End Sub

  Private Sub New()
    ' prevent direct creation
    MarkAsChild()
  End Sub

End Class

File: Csla20Test/Csla.Test/Basic/Children.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Data;

namespace Csla.Test.Basic
{
    [Serializable()]
    public class Children : BusinessListBase<Children, Child>
    {
        public void Add(string data)
        {
            Console.WriteLine("Adding Child to Children");
            this.Add(Child.NewChild(data));
            Console.WriteLine("\tNum children: {0}", this.Items.Count);
        }


        internal static Children NewChildren()
        {
            return new Children();
        }

        internal static Children GetChildren(IDataReader dr)
        {
            //todo: load child data
            return null;
        }

        internal void Update(IDbTransaction tr)
        {
            foreach (Child child in this)
            {
                child.Update(tr);
            }
        }

        private Children()
        {
            //prevent direct creation
            this.MarkAsChild();
        }
    }
}
